### ISSUE TITLE
Go to definition in VS Code

### ISSUE THREAD

**Issue by s-khechnev** [ocaml/ocaml#13022]
How can I use the "go to definition" feature in VS Code while working with the OCaml compiler's code base? Is it possible?

**Comment by gasche** [comment]
Yes, see [HACKING.adoc > using Merlin](https://github.com/ocaml/ocaml/blob/trunk/HACKING.adoc#using-merlin).

**Comment by Kakadu** [comment]
> Yes, see [HACKING.adoc > using Merlin](https://github.com/ocaml/ocaml/blob/trunk/HACKING.adoc#using-merlin).

Actually, I thought that answer is no https://discord.com/channels/436568060288172042/436568060288172044/1185917009129648128

> [3:10 PM]Kakadu: I have troubles building trunk compiler via dune. It complains about Error: Wrong arity for builtin primitive "%resume" and type error in 'toplevel/topcommon.ml'. Worth opening an issue?
[3:12 PM]Kakadu: Bulding a tag 5.1.1 via dune is troublesome too. A few warnings-as-errors, and Unbound module Compression
[3:13 PM]Kakadu: 5.1.0 is fine
[3:18 PM]Kakadu: OK, 5.1.0 is not fine, because runtime/ can't be built. (runtime/Makefile is missing) (edited)
[3:37 PM]octachron: In short, dune is not supported for building the compiler.
[3:50 PM]Kakadu: Is it fundamental incompatibility with dune, or just a lack of human resources?
[4:02 PM]octachron: A little of both

**Comment by Octachron** [comment]
This is an answer to a different question. To enable merlin support, it is sufficient to build `cmt` and `cmi` files for the compiler. Building the whole compiler is unnecessary.

**Comment by gasche** [comment]
In other words: the dune build system is a hack that works more or less. It never works for building compiler binaries, but it usually works well enough to provide Merlin support for the sources in most of the codebase, and it is the easiest known way to get such Merlin support.
